eureka/S5_lightcurve_fitting/fitters.py

```diff
@@ -198,63 +229,260 @@ def emceefitter(lc, model, meta, log, **kwargs):
         state issues.
 
     """
-    if not hasattr(meta, 'lsq_first') or meta.lsq_first:
-        # Only call lsq fitter first if asked or lsq_first option wasn't passed (allowing backwards compatibility)
-        log.writelog('\nCalling lsqfitter first...')
-        # RUN LEAST SQUARES
-        lsq_sol = lsqfitter(lc, model, meta, log, calling_function='emcee_lsq', **kwargs)
-
-        # SCALE UNCERTAINTIES WITH REDUCED CHI2
-        if meta.rescale_err:
-            lc.unc *= np.sqrt(lsq_sol.chi2red)
-    else:
-        lsq_sol = None
-    
     # Group the different variable types
     freenames, freepars, prior1, prior2, priortype, indep_vars = group_variables(model)
+    if hasattr(meta, 'old_fitparams') and meta.old_fitparams is not None:
+        freepars = load_old_fitparams(meta, log, lc.channel, freenames)
+    ndim = len(freenames)
+
+    if hasattr(meta, 'old_chain') and meta.old_chain is not None:
+        pos, nwalkers = start_from_oldchain_emcee(meta, log, ndim, lc.channel, freenames)
+    else:
+        if not hasattr(meta, 'lsq_first') or meta.lsq_first:
+            # Only call lsq fitter first if asked or lsq_first option wasn't passed (allowing backwards compatibility)
+            log.writelog('\nCalling lsqfitter first...')
+            # RUN LEAST SQUARES
+            lsq_sol = lsqfitter(lc, model, meta, log, calling_function='emcee_lsq', **kwargs)
+
+            freepars = lsq_sol.fit_params
+
+            # SCALE UNCERTAINTIES WITH REDUCED CHI2
+            if meta.rescale_err:
+                lc.unc *= np.sqrt(lsq_sol.chi2red)
+        else:
+            lsq_sol = None
+        pos, nwalkers = initialize_emcee_walkers(meta, log, ndim, lsq_sol, freepars, prior1, prior2, priortype)
+
+    start_lnprob = lnprob(np.median(pos, axis=0), lc, model, prior1, prior2, priortype, freenames)
+    log.writelog(f'Starting lnprob: {start_lnprob}', mute=(not meta.run_verbose))
+
+    # Initialize tread pool
+    if hasattr(meta, 'ncpu') and meta.ncpu > 1:
+        pool = Pool(meta.ncpu)
+    else:
+        pool = None
+    
+    # Run emcee burn-in
+    sampler = emcee.EnsembleSampler(nwalkers, ndim, lnprob, args=(lc, model, prior1, prior2, priortype, freenames),
+                                    pool=pool)
+    log.writelog('Running emcee burn-in...')
+    state = sampler.run_mcmc(pos, meta.run_nsteps, progress=True)
+    # # Log some details about the burn-in phase
```

Comment:
So I'm conflicted about how we should be running emcee. Either we can run one single chain like this which means that the autocorrelation time will always be either very bad or emcee won't even be able to calculate it, or we can do a sampler.reset() after the burn-in and before the production run which allows some more logging opportunities and allows a much more meaningful autocorrelation time to be calculated for the production run but comes at the risk of emcee throwing a warning that the walkers may not be linearly independent (an error I've encountered before when trying to resume emcee runs). I commented out the code for now since I wanted parallelized emcee and dynesty to get merged sooner than later without risking emcee fits breaking, but I left the code there so that I can run some more thorough tests once this data challenge is over to decide which technique we should use

Reply:
Ok, makes sense to keep it commented.